admin_guide/advanced_scheduling/pod_affinity.adoc

```diff
@@ -0,0 +1,150 @@
+
+* Pod affinity tells the scheduler to locate a new pod on the same node as pods with the same labels.
+* Pod anti-affinity prevents the scheduler from locating a new pod on the same node as pods with the same labels.
+
+There are currently two types of pod affinity rules, _required_ and _preferred_. 
+
+Required rules *must* be met before a pod can be scheduled on a node. Preferred rules specify that, if the rule is met, the scheduler will try to enforce the rules, but will not guarantee enforcement.
+
+[[admin-guide-sched-affinity-pod-config]]
+== Configuring Pod Affinity/Anti-Affinity
+
+Pod affinity/anti-affinity is configured through the pod specification file. You can specify a xref:admin-guide-sched-affinity-config-req[required rule], a xref:admin-guide-sched-affinity-config-pref[preferred rule], or both. If you specify both, the node must first meet the required rule, then meet the preferred.
+
+The following example shows a pod specification configured for pod affinity and anti-affinity. 
+
+The pod affinity rule says that the pod can schedule onto a node only if that node has at least one already-running pod with a label that has the key `security` and value `S1`. The pod anti-affinity rule says that the pod prefers to not schedule onto a node if that node is already running a pod with label having key `security `and value `S2`. 
+
+.Example pod config file with pod affinity and anti-affinity
+====
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: with-pod-affinity
+spec:
+  affinity:
+    podAffinity: <1>
+      requiredDuringSchedulingIgnoredDuringExecution: <2>
+      - labelSelector:
+          matchExpressions:
+          - key: security <3>
+            operator: In <4>
+            values:
+            - S1 <5>
+        topologyKey: failure-domain.beta.kubernetes.io/zone
+    podAntiAffinity: <6>
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: security <3>
+              operator: In <4>
+              values:
+              - S2 <5>
+          topologyKey: kubernetes.io/hostname
+  containers:
+  - name: with-pod-affinity
+    image: gcr.io/google_containers/pause:2.0
+
+----
+====
+<1> Stanza to configure pod affinity.
+<2> Creates a required rule.
+<3> The key that must be matched to apply the rule. 
+<3> The xref:../../rest_api/openshift_v1.html#unversioned-labelselectorrequirement[operator] represents the relationship between the key and a set of values.  Can be `In`, `NotIn`, `Exists`, or `DoesNotExist`.
+<5> The value that must be matched to apply the rule.
+<4> Creates a preferred rule.
+
+
+[NOTE]
+====
+If labels on a node change at runtime such that the affinity rules on a pod are no longer met, the pod will still continue to run on the node. 
```

Comment:
As above only true for *ignoredDuringExecution, not for *requiredDuringExecution (not yet implemented).